feat: move sales channel default data to fixtures (#305)

DanieliMi opened the pull request:
Moving the logic from https://github.com/shopware/acceptance-test-suite/blob/ba108ea999e25e825c46ec308202b247d077c089/src/fixtures/DefaultSalesChannel.ts#L52-L59 to new fixtures within ShopwareDataFixtures for better overwriting of default data.

pweyck commented on src/services/ShopwareDataHelpers.ts:
```diff
@@ -105,27 +105,6 @@ export const getPaymentMethodId = async (adminApiContext: AdminApiContext, handl
- * Gives the default shipping method back called Standard
- * @param adminApiContext - An AdminApiContext entity
- */
-export const getDefaultShippingMethodId = async (adminApiContext: AdminApiContext): Promise<string> => {
```
We should not removed it but just deprecate this method (otherwise it would be a break) and create the more general `getShippingMethodId`, that takes the name as the parameter.

DanieliMi replied:
Good point! I will change that and even use it in the fixture. Since it is a single fixture which we can override independently I see no problem in using the helper.

pweyck commented on src/fixtures/ShopwareDataFixtures.ts:
```diff
@@ -0,0 +1,36 @@
+
+    ShippingMethod: [
+        async ({ AdminApiContext }, use) => {
+            const resp = await AdminApiContext.post('search/shipping-method', {
```
Call the new method with `'Standard'` here

pweyck commented on src/fixtures/ShopwareDataFixtures.ts:
```diff
@@ -0,0 +1,36 @@
+
+export const test = base.extend<NonNullable<unknown>, FixtureTypes>({
+
+    ShippingMethod: [
```
For `PaymentMethod` we could do the same

DanieliMi replied:
Yes, I intend to change all of https://github.com/shopware/acceptance-test-suite/blob/ba108ea999e25e825c46ec308202b247d077c089/src/fixtures/DefaultSalesChannel.ts#L52-L59
to an fixture. Then you can overwrite them as needed.

pweyck reviewed (changes requested):
In general we should keep the helper functions, because they are useful in other contexts, but discourage (and eventually remove) the use of the ones which assume some data exists like `getDefaultShippingMethodId`

DanieliMi commented:
Thank you for having a look and the feedback!

> In general we should keep the helper functions, because they are useful in other contexts, but discourage (and eventually remove) the use of the ones which assume some data exists like `getDefaultShippingMethodId`

I just had a deeper look in how the helpers are used otherwise (I just quickly scanned it before when I made the suggestion) and I do agree. Most of them are used during data creation and the methods have parameters with which you  can easily provide the data you want. I'll then only change the ones that provide fixed default data in a fixture.

DanieliMi committed:
feat: move sales channel default data to fixtures (#305)

...move shipping method to fixture
...move payment method to fixture
...move language to fixture
...move currency to fixture
...move tax to fixture
...move country to fixture
...move snippet set to fixture
...move theme to fixture

DanieliMi committed:
Merge branch 'shopware:trunk' into 305/move-data-helpers-to-fixture

DanieliMi commented:
Hi @pweyck, I've done the requested changes and also moved the remaining default data logic. Can you have a look again?

DanieliMi commented:
Do you have any idea why the two tests fail?

frobel commented:
> Do you have any idea why the two tests fail?

1. Your second commit message is not using our conventional commit policy. 
2. I will rerun the pipeline. Maybe there was a temporarily issue in the API.

DanieliMi commented:
> > Do you have any idea why the two tests fail?
> 
> 1. Your second commit message is not using our conventional commit policy.
> 2. I will rerun the pipeline. Maybe there was a temporarily issue in the API.

@frobel I presume you mean b4415946986efd11c58d05e2bff30876d47d63e4. Can you explain to me what does match the conventional commit policy so I can fix it? I have read it and thought it does comply. The other commit is just a merge commit from updating the branch via GitHub directly.

pweyck commented:
> > > Do you have any idea why the two tests fail?
> > 
> > 
> > 
> > 1. Your second commit message is not using our conventional commit policy.
> > 2. I will rerun the pipeline. Maybe there was a temporarily issue in the API.
> 
> @frobel I presume you mean [b441594](https://github.com/shopware/acceptance-test-suite/commit/b4415946986efd11c58d05e2bff30876d47d63e4). Can you explain to me what does match the conventional commit policy so I can fix it? I have read it and thought it does comply. The other commit is just a merge commit from updating the branch via GitHub directly.

I've used squash merge, in that case it does not matter, because only the title is used for the commit message

frobel commented:
@DanieliMi we released your change in the ATS version 11.9.0

DanieliMi commented:
Ah alright :)  
Thank you very much!
